### PR TITLE
[7.5.1][CDE] ApiRGBColor.ToJSON check

### PR DESCRIPTION
**Pull request by Yar04ek**
**Functionality that needs to be tested**
1. ApiRGBColor.prototype.ToJSON; _on review_

### CODE REVIEW

File: js/docx/smoke/api_rgb_color/to_json.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
+builder.CreateFile("docx");
+    let oDocument = Api.GetDocument();
+    let oParagraph = oDocument.GetElement(0);
+    let oRGBColor = Api.CreateRGBColor(255, 111, 61);
+    let jsonRGBColor = oRGBColor.ToJSON();
+        oParagraph.AddText(jsonRGBColor);
+GlobalVariable["JSON_RGBColor"] = jsonRGBColor;
+builder.CloseFile();
+
```
Comment (askonev) [inline js/docx/smoke/api_rgb_color/to_json.js:9]: I think it's unnecessary blanks. One empty line is enough

File: js/docx/smoke/api_rgb_color/to_json.js [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+builder.CreateFile("docx");
+    let oNewDocument = Api.GetDocument();
+    let jsonRGBColor = GlobalVariable["JSON_RGBColor"];
+    let oParagraphRGBColor = oNewDocument.GetElement(0);
+        oParagraphRGBColor.AddText(jsonRGBColor);
+
+    let oRGBColorFromJSON = Api.FromJSON(jsonRGBColor);
+    let oParagraphRGBColorObject = Api.CreateParagraph();
+        oParagraphRGBColorObject.AddDrawing(oRGBColorFromJSON);
```
Comment (askonev) [inline js/docx/smoke/api_rgb_color/to_json.js:22]: RGBColor is not a drawing! If you manage to add something to a paragraph using AddDrawing method passing RGBCorol, it is a bug. 

RGBColor is used as part of the fill

[Study the documentation for fill obj CreateSolidFill](https://api.onlyoffice.com/docbuilder/textdocumentapi/api/createsolidfill)

File: js/docx/smoke/api_rgb_color/to_json.js [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+        oNewDocument.Push(oParagraphRGBColorObject);
+
+    let oGs1 = Api.CreateGradientStop(oRGBColorFromJSON, 0);
+    let oGs2 = Api.CreateGradientStop(Api.CreateRGBColor(255, 164, 101), 100000);
+    let oFill = Api.CreateLinearGradientFill([oGs1, oGs2], 5400000);
+    let oStroke = Api.CreateStroke(0, Api.CreateNoFill());
+    let oDrawingFromJSON = Api.CreateShape("rect", 5930900, 395605, oFill, oStroke);
```
Comment (askonev) [inline js/docx/smoke/api_rgb_color/to_json.js:29]: I think it's just a Drawing object. It's not derived from JSON

File: js/docx/smoke/api_rgb_color/to_json.js [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+    let bWriteNumberings = false;
+    let bWriteStyles = true;
+    let jsonDrawingFromJSON = oDrawingFromJSON.ToJSON(bWriteNumberings, bWriteStyles);
+
+    let oParagraphDrawing = Api.CreateParagraph();
+        oParagraphDrawing.AddText(jsonDrawingFromJSON);
+        oNewDocument.Push(oParagraphDrawing);
+
+    let oDrawingFromJSONObj = Api.FromJSON(jsonDrawingFromJSON);
```
Comment (askonev) [inline js/docx/smoke/api_rgb_color/to_json.js:38]: I don't think you should create a Drawing object from json at all.

You have already set Drawing json that is enough

File: spec/docx/smoke/api_rgb_color_spec.rb [hunk 5]
```diff
@@ -6,4 +6,11 @@
     docx = builder.build_and_parse('js/docx/smoke/api_rgb_color/get_class_type.js')
     expect(docx.elements[1].nonempty_runs.first.text).to eq('Class Type = rgbColor')
   end
+
+  it 'ApiRGBColor | ToJSON method' do
+    docx = builder.build_and_parse('js/docx/smoke/api_rgb_color/to_json.js')
+    first_paragraph_text = '{"rgba":{"red":0,"green":0,"blue":0,"alpha":255},"color":{"rgba":{"red":255,"green":111,"blue":61,"alpha":255},"type":"srgb"},"type":"uniColor"}'
```
Comment (askonev) [inline spec/docx/smoke/api_rgb_color_spec.rb:12]: When reading data from a document, use JSON.parse() to get a ruby object and interact with it.

**Review by askonev — CHANGES_REQUESTED** [review]

File: spec/docx/smoke/api_rgb_color_spec.rb [hunk 6]
```diff
@@ -9,8 +9,10 @@
 
   it 'ApiRGBColor | ToJSON method' do
     docx = builder.build_and_parse('js/docx/smoke/api_rgb_color/to_json.js')
-    first_paragraph_text = '{"rgba":{"red":0,"green":0,"blue":0,"alpha":255},"color":{"rgba":{"red":255,"green":111,"blue":61,"alpha":255},"type":"srgb"},"type":"uniColor"}'
+    expected_text = '{"rgba":{"red":0,"green":0,"blue":0,"alpha":255},"color":{"rgba":{"red":255,"green":111,"blue":61,"alpha":255},"type":"srgb"},"type":"uniColor"}'
```
Comment (askonev) [inline spec/docx/smoke/api_rgb_color_spec.rb:12]: I meant that you can read the text directly from the document.
```ruby
docx.elements[0].nonempty_runs.first.text
```
Or create a ruby JSON object right away 
```ruby
json = JSON.parse(docx.elements[0].nonempty_runs.first.text)
```

**Review by askonev — CHANGES_REQUESTED** [review]

File: spec/docx/smoke/api_rgb_color_spec.rb (hunk 6 above)
Comment (Yar04ek) [reply spec/docx/smoke/api_rgb_color_spec.rb]: fixed in new commit

Comment (Yar04ek) [reply spec/docx/smoke/api_rgb_color_spec.rb]: fixed in new commit

**Pull request closed without merging**